components/psst/browser/content/psst_scripts_result_handler.h

```diff
@@ -0,0 +1,88 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_PSST_BROWSER_CONTENT_PSST_SCRIPTS_RESULT_HANDLER_H_
+#define BRAVE_COMPONENTS_PSST_BROWSER_CONTENT_PSST_SCRIPTS_RESULT_HANDLER_H_
+
+#include <memory>
+#include <string>
+
+#include "base/component_export.h"
+#include "brave/components/script_injector/common/mojom/script_injector.mojom.h"
+#include "content/public/browser/web_contents.h"
+#include "mojo/public/cpp/bindings/associated_remote.h"
+
+class PrefService;
+
+namespace psst {
+
+class MatchedRule;
+
+class COMPONENT_EXPORT(PSST_BROWSER_CONTENT) PsstScriptsHandler {
+ public:
+  using InsertScriptInPageCallback = base::OnceCallback<void(::base::Value)>;
+
+  PsstScriptsHandler(const PsstScriptsHandler&) = delete;
+  PsstScriptsHandler& operator=(const PsstScriptsHandler&) = delete;
+
+  PsstScriptsHandler();
+  virtual ~PsstScriptsHandler();
+
+  virtual void Start() = 0;
+
+ private:
+  virtual void InsertUserScript(const std::optional<MatchedRule>& rule) = 0;
```

Comment:
I don't understand why you have private pure virtual methods in a pure abstract class. This would make sense to me if `Start` had an implementation that called these and they were overridden in the subclass, but that's not the case here. Also why do we need an impl/interface here in the first place? There is only one subclass and the impl and interface are in the same file so it's not reducing any dependencies. Normally if you wanted to split them into interface/impl you'd have something like `PsstScriptHandler::Create` and define it in `psst_scripts_handler_impl.cc` so you only expose the interface outside of the component, but you have the impl here and you are even exporting it which makes this even more confusing

Reply:
also the filename should be `psst_scripts_handler.h` to match the class name